src/main/java/com/poiji/util/Casting.java

```diff
@@ -128,13 +128,24 @@ private Date dateValue(String value, PoijiOptions options) {
         if (options.getDateRegex() != null && !value.matches(options.getDateRegex())) {
             return defaultDate(options);
         } else {
-            try {
-                final SimpleDateFormat sdf = new SimpleDateFormat(options.datePattern());
-                sdf.setLenient(options.getDateLenient());
-                return sdf.parse(value);
-            } catch (ParseException e) {
-                return defaultDate(options);
+            String[] patterns = options.datePattern();
+            Date parsed = defaultDate(options);
+            for(String pattern:patterns) {
+                try {
+                    final SimpleDateFormat sdf = new SimpleDateFormat(pattern);
+                    //bug
```

Comment:
incase of parse date in lenient mode and datepattern mismatched,it will give me a strange date
(eg. 

> Wed Aug 10 00:00:00 PST 35
> C.E. 35-08-10T00:00:00.000-0800
> JulianCalendar$Date
> [CommnEar(C.E.) since 0000-12-30T16:47:04.192 local time

),the same issue reproduce in **DeserializersDateLenientTest** testcase.
i could't understood what is this.